### DISCUSSION TITLE
Is the compiler actively being maintained?

### DISCUSSION THREAD

**Discussion by nachumg** [SystemRDL/systemrdl-compiler#234] [Q&A] [answered]
I see the last change was in May 2024. https://github.com/SystemRDL/systemrdl-compiler/commits/main/

**Comment by amykyta3** [comment, marked as answer]
Thanks for asking! Indeed the project has been a bit quiet lately but it is certainly not abandoned. I have taken a little bit of a sabbatical from working on it due to some other [volunteer work](https://ecosystemsentinels.org/testing-technology-with-the-woodland-park-zoo/) I have been involved with this past year, but that is wrapping up and I am already working through some of the PeakRDL backlog. The recent growth in popularity of the tools in the past year has been exciting and humbling to say the least!

As to the compiler, the v1.x major version lineage is not likely to get many changes aside from bugfixes. I am planning development of a major v2 release which will pull in some of the major feature requests as well as some long-overdue clean-up and usability enhancements.

**Comment by nachumg** [comment]
Great Alex!

Should I be sending pull requests for the main branch, is do you want to create a v2 branch?

**Reply by amykyta3** [reply]
Recommend discussing pull requests in issue tickets first. But they yes, PRs always go to main and I may choose to rebase elsewhere if needed.
